libs/application/templates/aosh/transfer-of-machine-ownership/src/fields/ReviewOperatorRepeater/index.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { FC, useState } from 'react'
+import {
+  Box,
+  Button,
+  Divider,
+  GridColumn,
+  GridRow,
+  Text,
+} from '@island.is/island-ui/core'
+import { FieldBaseProps } from '@island.is/application/types'
+import { useLocale } from '@island.is/localization'
+import debounce from 'lodash/debounce'
+import { InputController } from '@island.is/shared/form-fields'
+import { ReviewScreenProps } from '../../shared'
+import { NationalIdWithName } from '../NationalIdWithName'
+import { error, information, review } from '../../lib/messages'
+import { UPDATE_APPLICATION } from '@island.is/application/graphql'
+import { useMutation } from '@apollo/client'
+
+export const ReviewOperatorRepeater: FC<
+  React.PropsWithChildren<FieldBaseProps & ReviewScreenProps>
+> = ({ setStep, setBuyerOperator, buyerOperator = {}, ...props }) => {
+  const { application } = props
+  const { locale, formatMessage } = useLocale()
+  const [updateApplication] = useMutation(UPDATE_APPLICATION)
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(
+    undefined,
+  )
+  const [genericErrorMessage, setGenericErrorMessage] = useState<
+    string | undefined
+  >(undefined)
+  const [name, setName] = useState<string | null>(buyerOperator?.name || null)
+  const [nationalId, setNationalId] = useState<string | null>(
+    buyerOperator?.nationalId || null,
+  )
+  const [email, setEmail] = useState<string | null>(
+    buyerOperator?.email || null,
+  )
+  const [phone, setPhone] = useState<string | null>(
+    buyerOperator?.phone || null,
+  )
+
+  const handleClear = async () => {
+    setName(null)
+    setNationalId(null)
+    setEmail('')
+    setPhone('')
+    setBuyerOperator && setBuyerOperator({})
+    buyerOperator = {}
+    const res = await updateApplication({
```

Comment:
When we have to resort to use updateApplication with in the applications I always hear warning bells in my head. Cant we find a way to use the formbuilder to reach the similiar flow for this review screen? What do you think?

Reply:
Removed updateApplication and used different method to get the same results
Thank you!